docs/persistent_compilation_cache.md

```diff
@@ -67,8 +67,135 @@ Cloud Storage (GCS) bucket. We recommend the following configuration:
 Assuming that `gs://jax-cache` is the GCS bucket, set `cache-location` as
 follows:
 
-```
+```python
 import jax
 
 jax.config.update("jax_compilation_cache_dir", "gs://jax-cache")
 ```
+
+## How it works
+
+The JAX compilation cache works by hashing a number of parameters to create a signature for a compiled function. These are:
+
+* The computation performed by the function captured by the non-optimized HLO of the JAX function being hashed
+
+* The jaxlib version 
+
+* Relevant XLA compilation flags
+
+* Device configuration captured in general, by the number of devices and the topology of the devices. 
+  Currently for GPUs, the topology only contains a string representation of the GPU name
+
+* Compression algorithm used to compress the compiled executable
+
+* Any custom hooks
+
+When the signature for a function created using the parameters above matches 
+that of a compiled function in the persistent cache the function will not be compiled, 
+but will just be read and deserialized from the persistent cache. 
+
+### Caching thresholds
+
+There are two thresholds that control whether JAX caches an executable. 
+These are `jax_persistent_cache_min_entry_size_bytes` and `jax_persistent_cache_min_compile_time_secs`. 
+Only are at least `jax_persistent_cache_min_entry_size_bytes` large and take `jax_persistent_cache_min_compile_time_secs` 
+long to compile will be cached. To cache every executable that is compiled, 
+you can set the former to -1 and the latter to 0 as follows:
+
+```python
+import jax
+
+jax.config.update("jax_persistent_cache_min_entry_size_bytes", -1)
+jax.config.update("jax_persistent_cache_min_compile_time_secs", 0)
+```
+
+## Different Runtimes
+
+Below we outline some observed behavior of the persistent compilation cache 
+in a variety of different runtimes as it relates to which processes write to the cache.
+
+### Single node with single process and single device
+
+This is the simplest runtime and the only process that compiles and writes to the compilation cache is the singular process. 
+The number of devices does not matter to the cache key in this setup, only the type of device does. 
+
+### Single node with single process and multiple devices
+
+Once again the only process that compiles and writes to the compilation cache is the singular proess. 
+The difference between this setup and the previous is that now the number of devices matters in addition to the type of device. 
+
+### Multiple process and multiple devices (on either single or multiple nodes)
```

Comment:
There is a jax flag `jax_share_binary_between_hosts` and an issue https://github.com/google/jax/issues/18819 which might be related to this topic.